src/Symfony/Component/DependencyInjection/Tests/Issues49591Test.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\DependencyInjection\Argument\ServiceLocatorArgument;
+use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Dumper\PhpDumper;
+use Symfony\Component\DependencyInjection\Reference;
+
+class Issues49591Test extends TestCase
+{
+    public function testServices()
+    {
+        $containerBuilder = new ContainerBuilder();
+
+        $containerBuilder->register('connection', Connection::class)
+            ->addArgument(new ServiceLocatorArgument(new TaggedIteratorArgument('subscriber', needsIndexes: true)))
+            ->setPublic(true);
+
+        $containerBuilder->register('repository', Repository::class);
+
+        $containerBuilder->register('session', Session::class)
+            ->setFactory([new Reference('repository'), 'login'])
+            ->addArgument(new Reference('connection'))
+            ->setPublic(true);
+
+        $containerBuilder->register('subscriber', Subscriber::class)
+            ->addArgument(new Reference('session'))
+            ->addTag('subscriber')
+            ->setPublic(true);
+
+        $containerBuilder->compile();
+
+        $dumper = new PhpDumper($containerBuilder);
+        $dump = $dumper->dump(['class' => 'Symfony_DI_PhpDumper_Issues49591']);
+
+        self::assertStringEqualsFile(__DIR__.'/Fixtures/php/services_issues49591.php', $dump);
```

Comment:
This is the assertion that fails currently. But as far as I understood the issue, the actual issue is reproduced by the following assertion. Do we need this fixture anyway? Doesn't it just make the test brittle?

Reply:
I've removed the assertion & updated the dump for version 6.3 (I originally generated it with 6.2).